serif_shared/src/commonMain/kotlin/xyz/room409/serif/serif_shared/States.kt

```diff
@@ -78,44 +78,120 @@ class MatrixRooms(private val msession: MatrixSession, val message: String) : Ma
         return MatrixLogin("Closing session, returning to the login prompt for now\n", MatrixClient())
     }
 }
-data class SharedUiMessage(
-    val sender: String,
-    val message: String,
-    val id: String,
-    val timestamp: Long,
-    val url: String? = null
+open class SharedUiMessage(
+    open val sender: String,
+    open val message: String,
+    open val id: String,
+    open val timestamp: Long,
+    open val replied_event: String = "",
+    open val edited_event: String = ""
```

Comment:
Perhaps these should be pointers to the SharedUiMessage they refer to?

Reply:
Not sure it makes sense to have edited_event be a pointer as it was just used to specifically mark the message with an "(edited)" tag and in the edits map construction. Replies it makes more sense, though what if we have a timeline state with the reply but not the original message in memory? I was going to use the field to find the message we want to jump to on message clicks in the future once we got some progress on view ports.

Reply:
That's fair, that makes sense

Reply:
I did remove the now useless edited_event field though.